pkg/output/output.go

```diff
@@ -55,31 +55,33 @@ func EnsureOutputPath(path string, deleteIfExists bool) error {
 			return os.Mkdir(path, 0700)
 		}
 		return err
-	} else {
+	}
+
+	if !info.IsDir() {
+		return fmt.Errorf("output destination must be a directory: '%s'", path)
+	}
+
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		return fmt.Errorf("failed to get contents of output directory '%s': %w", path, err)
+	}
+
+	if len(entries) != 0 {
 		if deleteIfExists {
 			err = os.RemoveAll(path)
 			if err != nil {
 				return fmt.Errorf("error while deleting the output path '%s': %w", path, err)
 			}
-
-			err = os.Mkdir(path, 0700)
-			if err != nil {
-				return fmt.Errorf("failed to create output directory '%s': %w", path, err)
-			}
 		} else {
-			if !info.IsDir() {
-				return fmt.Errorf("output destination must be a directory: '%s'", path)
-			}
-			entries, err := os.ReadDir(path)
-			if err != nil {
-				return fmt.Errorf("failed to get contents of output directory '%s': %w", path, err)
-			}
-			if len(entries) != 0 {
-				return fmt.Errorf("output directory %s is not empty", path)
-			}
+			return fmt.Errorf("output directory %s is not empty", path)
```

Comment:
This should be after the `if deleteIfExists{}`

Reply:
it is after the `deleteIfExists`. or do you mean to just leave the else statement out?

Reply:
No this fine. I was reviewing in between updates.